### PR TITLE
Fix C API test ie_core_register_plugins error

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Triger plugin load after register in test

### Tickets:
 - 109148

### CODE REVIEW

File: src/bindings/c/tests/test_model_repo.hpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #include "openvino/core/visibility.hpp"
 #include "openvino/pass/manager.hpp"
 #include "openvino/util/file_util.hpp"
+#include "common_test_utils/file_utils.hpp"
```
Comment (ilyachur) [inline src/bindings/c/tests/test_model_repo.hpp:15]: Please fix code style

File: src/bindings/c/tests/test_model_repo.hpp [hunk 2]
```diff
@@ -80,12 +68,21 @@ inline std::string generate_test_xml_file() {
 
     plugin_xml_file << "<ie>\n";
     plugin_xml_file << "    <plugins>\n";
-    plugin_xml_file << "        <plugin location=\"" << libraryname << "\" name=\"CUSTOM\">\n";
+    plugin_xml_file << "        <plugin location=\"";
+    plugin_xml_file << CommonTestUtils::getExecutableDirectory();
+    plugin_xml_file << ov::util::FileTraits<char>::file_separator;
+    plugin_xml_file << ov::util::FileTraits<char>::library_prefix();
+    plugin_xml_file << "mock_engine";
```
Comment (ilyachur) [inline src/bindings/c/tests/test_model_repo.hpp:75]: Can we add dependency on mock_engine for c tests in CMake?

File: src/bindings/c/tests/CMakeLists.txt [hunk 3]
```diff
@@ -3,6 +3,8 @@
 #
 
 # OpenVINO Legacy C API test sample
+set(DEPENDENCIES mock_engine)
```
Comment (ilyachur) [inline src/bindings/c/tests/CMakeLists.txt:6]: I think it is not correct command. You need to use `add_dependencies()` command

Comment (ilyachur) [reply src/bindings/c/tests/CMakeLists.txt:6]: And do we still need to have dependencies on hw plugins below?

Comment (zhaixuejun1993) [reply src/bindings/c/tests/CMakeLists.txt:6]: > add_dependencies()
Will correct this.


> And do we still need to have dependencies on hw plugins below?
I think u are right, the hw plugins dependencies will be removed

**Comment by zhaixuejun1993** [comment]
Closed! Using the new PR replace. https://github.com/openvinotoolkit/openvino/pull/17800

**Pull request closed without merging**